main.cpp

```diff
@@ -26,14 +26,20 @@ void testPairToNumber(
     assert(pairNumber == expectedPairNumber);
 }
 
+void testShowColorCodeReference(int (*func)())
+{
+    int maxPairCount = func();
+    assert(maxPairCount == TelCoColorCoder::numberOfMajorColors * TelCoColorCoder::numberOfMinorColors);
```

Comment:
It's a good test - checks if all the combinations were covered; it also checks that the last number is good. So if we initialize pairNumber with 0 by mistake inside the function, it will fail.
Think of a way to test the content of the report as well. We will discuss more tomorrow.